Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,316 @@
+    private static let apiKey = "DUMMY_KEY"
+    private static let maxBandwidth = 2048
+    private static let maxDiskSpace = 4096
+    private static let clientId = AppInfo.baseBundleIdentifier
+    private static let version = AppInfo.appVersion
+    
+    private let baseURL = "https://safebrowsing.brave.com"
+    private let session = URLSession(configuration: .ephemeral)
+    private let database = SafeBrowsingDatabase()
+    
+    public static let shared = SafeBrowsingClient()
+    
+    private init() {
+    }
+    
+    func find(_ hashes: [String], _ completion: @escaping (_ isSafe: Bool, Error?) -> Void) {
+        let discoveredHashes = database.find(hashes)
+        if discoveredHashes.isEmpty {
+            return completion(true, nil)
+        }
+        
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let threatTypes: [ThreatType] = [.malware,
+                                         .socialEngineering,
+                                         .unwantedSoftware,
+                                         .potentiallyHarmfulApplication]
+        
+        let platformTypes: [PlatformType] = [.any]
+        let threatEntryTypes: [ThreatEntryType] = [.url, .exe]
+        
+        let threatInfo = ThreatInfo(threatTypes: threatTypes,
+                                    platformTypes: platformTypes,
+                                    threatEntryTypes: threatEntryTypes,
+                                    threatEntries: discoveredHashes.map {
+                                        return ThreatEntry(hash: $0, url: nil, digest: nil)
+                                    }
+        )
+        
+        do {
+            let body = FindRequest(client: clientInfo, threatInfo: threatInfo)
+            let request = try encode(.post, endpoint: .fullHashes, body: body)
+            executeRequest(request, type: FindResponse.self) { response, error in
+                if let error = error {
+                    return completion(false, error)
+                }
+                
+                if let response = response {
+                    if !response.matches.isEmpty {
+                        return completion(false, nil)
+                    }
+                }
+                
+                completion(true, nil)
+            }
+        } catch {
+            completion(false, error)
+        }
+    }
+    
+    func fetch(_ completion: @escaping (Error?) -> Void) {
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let constraints = Constraints(maxUpdateEntries: UInt32(SafeBrowsingClient.maxBandwidth),
+                                      maxDatabaseEntries: UInt32(SafeBrowsingClient.maxDiskSpace),
+                                      region: "US",
```

Comment:
Region can be non-us as well, I think `Locale.current.regionCode` will help here.